src/github/github.py

```diff
@@ -0,0 +1,192 @@
+import os
+from dataclasses import dataclass
+from json import dumps
+from typing import Any
+
+import requests
+from constants import github_org_name
+from requests import Response
+
+
+@dataclass
+class GithubUser:
+    username: str
+    id: int
+    name: str
+
+@dataclass
+class GithubTeam:
+    id: int
+    url: str
+    name: str
+    # slugs are the space-free special names that github likes to use
+    slug: str
+
+async def _github_request_get(
+        url: str,
+        token: str
+) -> Response:
+    result = requests.get(
+        url,
+        headers={
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {token}",
+            "X-GitHub-Api-Version": "2022-11-28"
+        }
+    )
+    rate_limit_remaining = int(result.headers["x-ratelimit-remaining"])
+    if rate_limit_remaining < 50:
+        return None
+
+    return result
+
+async def _github_request_post(
+        url: str,
+        token: str,
+        post_data: Any
+) -> Response:
+    result = requests.post(
+        url,
+        headers={
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {token}",
+            "X-GitHub-Api-Version": "2022-11-28"
+        },
+        data=post_data
+    )
+    rate_limit_remaining = int(result.headers["x-ratelimit-remaining"])
+    if rate_limit_remaining < 50:
+        return None
```

Comment:
Remember to update the return type if it's optional
```suggestion
) -> Response | None:
    result = requests.post(
        url,
        headers={
            "Accept": "application/vnd.github+json",
            "Authorization": f"Bearer {token}",
            "X-GitHub-Api-Version": "2022-11-28"
        },
        data=post_data
    )
    rate_limit_remaining = int(result.headers["x-ratelimit-remaining"])
    if rate_limit_remaining < 50:
        return None
```

Reply:
Resolved in 0ddf93e